### ISSUE TITLE
Handle unparseable UniqueID consistently

### ISSUE THREAD

**Issue by jlink** [junit-team/junit5#210]
Discovering tests via `UniqueIdSelector` and a string-based _unique ID_ that cannot be parsed by `UniqueId.parse()` is handled differently by each engine. Should be handled consistently in the launcher.

I see several options:

- Throw (subtype of) `JUnitException` when executing the request or when creating the selector instance
- Ignore selector and log with warning

I'd prefer the first one since a syntactically wrong unique ID can only come from a bug by users of the launcher API. It's like violating a precondition.

**Comment by marcphilipp** [comment]
Original issue: #171.

**Comment by marcphilipp** [comment]
Additional Proposals:
- Return a "AlwaysFailingTestDescriptor"/"DeadOnArrivalTestDescriptor"/"ErrorReportingTestDescriptor"
- Pass Object from Launcher into TestEngines to report unparsable unique IDs

**Comment by marcphilipp** [comment]
Since TestEngines now receive an already parsed UniqueId, I think this issue can be closed. The error reporting parts will be addressed in #242.

@junit-team/junit-lambda Thoughts?

**Comment by jlink** [comment]
Haven't read up on #242 but "unparsable" is different from "unresolvable". Unparsable is handled by parsing upfront. Unresolvable is not.

**Comment by marcphilipp** [comment]
Fair enough. It seems the issue title and description are out of sync. Would you like to rephrase it to say unresolvable in the description as well?

**Comment by jlink** [comment]
Should probably be a new issue.

**Comment by sbrannen** [comment]
FYI: I changed this issue's title to refer to "unparseable" unique IDs.

**Comment by sbrannen** [comment]
Note that consistent parsing of unique IDs was addressed in #810.

**Comment by sbrannen** [comment]
Closing this issue since it has been effectively superseded by #242 and #810.
